### PR TITLE
[aggregator] Fix simulation & view for delayed fields

### PR DESCRIPTION
**Pull request by georgemitenkov**
### Description

Previously, we have been tagging aggregator fields if the feature flag is enabled. This is not correct because it can be enabled but we run under non-capable context (i.e., outside of Block-STM). This PR fixes that by explicitly providing a boolean whether type layouts need to be tagged inside a VM.

### Test Plan

- Test for view function
- Test for simulation

### CODE REVIEW

**Review by igor-aptos — APPROVED** [review]

File: aptos-move/aptos-vm/src/aptos_vm.rs [hunk 1]
```diff
@@ -172,6 +172,22 @@ pub struct AptosVM {
 
 impl AptosVM {
     pub fn new(resolver: &impl AptosMoveResolver) -> Self {
+        Self::new_impl(resolver, resolver.is_delayed_field_optimization_capable())
+    }
+
+    pub(crate) fn new_with_delayed_fields(resolver: &impl AptosMoveResolver) -> Self {
+        Self::new_impl(resolver, true)
```
Comment (ziaptos) [inline aptos-move/aptos-vm/src/aptos_vm.rs:179]: Should we assert `resolver.is_delayed_field_optimization_capable())`

File: api/src/tests/simulation_test.rs [hunk 2]
```diff
@@ -82,3 +86,57 @@ async fn test_simulate_transaction_with_insufficient_balance() {
     let resp = simulate_aptos_transfer(&mut context, false, LARGE_TRANSFER_AMOUNT, 200).await;
     assert!(!resp[0]["success"].as_bool().is_some_and(|v| v));
 }
+
+#[tokio::test(flavor = "multi_thread", worker_threads = 2)]
```
Comment (ziaptos) [inline api/src/tests/simulation_test.rs:90]: Do we know what is this `worker_threads = 2` achieving?

File: api/goldens/aptos_api__tests__view_function__test_view_aggregator.json [hunk 3]
```diff
@@ -0,0 +1 @@
+["10"]
```
Comment (ziaptos) [inline api/goldens/aptos_api__tests__view_function__test_view_aggregator.json:1]: newline

**Review by ziaptos — APPROVED** [review]
Please address the comments before landing

File: api/src/tests/simulation_test.rs (hunk 2 above)
Comment (georgemitenkov) [reply api/src/tests/simulation_test.rs:90]: This makes sure your runtime is configured as multi-threaded with 2 threads to run async

File: aptos-move/aptos-vm/src/aptos_vm.rs [hunk 4]
```diff
@@ -172,6 +172,22 @@ pub struct AptosVM {
...
+    }
+
+    // Used by simulation & view functions.
+    fn new_no_delayed_fields(resolver: &impl AptosMoveResolver) -> Self {
+        Self::new_impl(resolver, false)
+    }
+
+    fn new_impl(
+        resolver: &impl AptosMoveResolver,
+        is_delayed_field_optimization_capable: bool,
```
[the remark this answers is not in the document]
Comment (georgemitenkov) [reply aptos-move/aptos-vm/src/aptos_vm.rs:189]: Good point!

File: aptos-move/aptos-vm/src/aptos_vm.rs (hunk 1 above)
Comment (georgemitenkov) [reply aptos-move/aptos-vm/src/aptos_vm.rs:179]: Use a single constructor instead

File: api/goldens/aptos_api__tests__view_function__test_view_aggregator.json (hunk 3 above)
Comment (georgemitenkov) [reply api/goldens/aptos_api__tests__view_function__test_view_aggregator.json:1]: golden tests compare the full file so new line breaks them

File: api/src/tests/move/pack_counter/sources/counter.move [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+        move_to(account, counter);
+    }
+
+    public entry fun increment_counter() acquires Counter {
+        let counter = &mut borrow_global_mut<Counter>(@addr).counter;
+        aggregator_v2::add(counter, 1);
+    }
+
+    #[view]
+    public fun get_counter_value(): u64 acquires Counter {
```
Comment (vusirikala) [inline api/src/tests/move/pack_counter/sources/counter.move:21]: `increment_and_get_counter_value`

File: aptos-move/aptos-debugger/src/aptos_debugger.rs [hunk 6]
```diff
@@ -78,7 +78,10 @@ impl AptosDebugger {
 
         // TODO(Gas): revisit this.
         let resolver = state_view.as_move_resolver();
-        let vm = AptosVM::new(&resolver);
+        let vm = AptosVM::new(
+            &resolver,
+            /*override_is_delayed_field_optimization_capable=*/ Some(false),
```
Comment (vusirikala) [inline aptos-move/aptos-debugger/src/aptos_debugger.rs:83]: Why do we need to disable aggregators here?

**Review by vusirikala — APPROVED** [review]

File: aptos-move/aptos-debugger/src/aptos_debugger.rs (hunk 6 above)
Comment (georgemitenkov) [reply aptos-move/aptos-debugger/src/aptos_debugger.rs:83]: This runs outside of Block-STM context

File: api/src/tests/move/pack_counter/sources/counter.move [hunk 7]
```diff
@@ -0,0 +1,26 @@
+module addr::counter {
```
Comment (gelash) [inline api/src/tests/move/pack_counter/sources/counter.move:1]: license headers?

Comment (georgemitenkov) [reply api/src/tests/move/pack_counter/sources/counter.move:1]: Do we need them for test sources as well?

**Pull request merged**